pkg/clone/replicate.go

```diff
@@ -157,7 +159,26 @@ func (cmd *Replicate) run(ctx context.Context) error {
 	}
 	if cmd.DoSnapshot {
 		go func() {
-			time.Sleep(cmd.DoSnapshotDelay)
+			logger := logrus.WithField("task", "snapshot")
+			for {
+				delay := cmd.DoSnapshotMaxReplicationLagCheckInterval
+				lag := replicator.GetReplicationLag()
+				if err != nil {
+					logger.WithError(err).Warnf("failed to check replication lag, checking again in %v", delay)
+					continue
+				}
+				if lag < cmd.DoSnapshotMaxReplicationLag {
```

Comment:
- [ ] if the lag is always >= cmd.DoSnapshotMaxReplicationLag, would it be an infinite loop?

Reply:
yeah it wouldn't start a snapshot until the lag is low enough. I think that's good?